### PR TITLE
setting uv_tcp_keepalive

### PR DESCRIPTION
**Pull request by olgavrou**
Regarding #429 

Setting TCP keepalive to 30 seconds (default on ubuntu 18.04 is 75 seconds). Expecting this to mitigate the issue seen in the ticket.

When running a client that connects with a CCF node via a firewall or a load balancer, if the connection is idle, the connection is considered to be closed by the client (client connection probably gets a RST but CCF side doesn't), where as CCF still holds a libuv descriptor for the connection. If the client then goes on to open a new connection every time this happens, we will run out of file descriptors on the CCF side, with no way of cleaning up for dead connections.

This could also happen in other cases of sudden network connectivity loss where the connection doesn't terminate properly.

Was able to reproduce this issue by running a CCF node on an Azure VM and running the python client on my desktop and having the client sleep for long intervals between sending transactions.

### CODE REVIEW

File: src/host/tcp.h [hunk 1]
```diff
@@ -190,6 +190,8 @@ namespace asynchost
         return false;
       }
 
+      uv_tcp_keepalive(&uv_handle, 1, 30);
```
Comment (eddyashton) [inline src/host/tcp.h]: We should check the return code of this (see http://docs.libuv.org/en/v1.x/guide/basics.html#error-handling).

**Comment by cimetrics** [comment]
images

File: src/host/tcp.h (hunk 1 above)
Comment (olgavrou) [reply src/host/tcp.h]: thanks!

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**